src/test/java/com/google/sps/NewMatchQueryServletTest.java

```diff
@@ -70,24 +79,36 @@ public void tearDown() throws Exception {
     }
 
     @Test
-    public void doPost_returnsNewMatches() throws IOException, ServletException {
+    public void doPostReturnError() throws IOException, ServletException {
         when(sessionContext.getLoggedInUser()).thenReturn(testUser);
-
-        MatchQuery matchQuery = new MatchQuery();
-        Collection<User> userSavedMatches = new ArrayList<User>();
-        Collection<User> answer = matchQuery.query(new MatchRequest(), userSavedMatches);
- 
+        when(abuseFeature.addRequest(Mockito.any(Date.class))).thenReturn(false);
+    
+      
         StringWriter StringWriter = new StringWriter();
         PrintWriter printWriter = new PrintWriter(StringWriter);
         when(response.getWriter()).thenReturn(printWriter);
 
         newMatchQueryServlet.doPost(request, response);
 
         Gson gson = new Gson();
-        String expected = gson.toJson(answer);
+        String expected = gson.toJson("/error.html");
         String result = StringWriter.getBuffer().toString().trim();
 
         printWriter.flush();
         Assert.assertEquals(expected, result);
     }
+
+    @Test
+    public void doPostAbuseFeatureNotInitialized() throws IOException, ServletException {
```

Comment:
L102-113: This is fine, I was just curious if you had any use-case/scenario in mind for the AbuseDetection module being null in matchQuery

Reply:
I don't really have a way to handle that scenario.